From 42517eb3b0228bbe8e6ad94b4df92a0fedc38ecb Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sat, 20 Apr 2024 12:27:03 +0300
Subject: [PATCH] tests: Clear scan cache in ap_hs20_anqp_invalid_gas_response

This is needed to avoid unexpected behavior if a previously executed
test case has left a BSS entry with Interworking emabled into the case.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 tests/hwsim/test_ap_hs20.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/hwsim/test_ap_hs20.py b/tests/hwsim/test_ap_hs20.py
index c1aebaa85..9cf2d9040 100644
--- a/tests/hwsim/test_ap_hs20.py
+++ b/tests/hwsim/test_ap_hs20.py
@@ -5758,6 +5758,7 @@ def test_ap_hs20_anqp_invalid_gas_response(dev, apdev):
     params['hessid'] = bssid
     hapd = hostapd.add_ap(apdev[0], params)
 
+    dev[0].flush_scan_cache()
     dev[0].scan_for_bss(bssid, freq="2412")
     hapd.set("ext_mgmt_frame_handling", "1")
 
-- 
2.43.0

